test/unit/org/apache/cassandra/db/ColumnFamilyStoreTest.java

```diff
@@ -266,12 +266,11 @@ public void testBackupAfterFlush() throws Throwable
         new RowUpdateBuilder(cfs.metadata(), 0, ByteBufferUtil.bytes("key2")).clustering("Column1").add("val", "asdf").build().applyUnsafe();
         cfs.forceBlockingFlush();
 
-        for (int version = 1; version <= 2; ++version)
+        for (SSTableReader liveSSTable : cfs.getLiveSSTables())
         {
-            Descriptor existing = new Descriptor(cfs.getDirectories().getDirectoryForNewSSTables(), KEYSPACE2, CF_STANDARD1, version,
-                                                 SSTableFormat.Type.BIG);
-            Descriptor desc = new Descriptor(Directories.getBackupsDirectory(existing), KEYSPACE2, CF_STANDARD1, version, SSTableFormat.Type.BIG);
-            for (Component c : new Component[]{ Component.DATA, Component.PRIMARY_INDEX, Component.FILTER, Component.STATS })
+            Descriptor existing = liveSSTable.descriptor;
+            Descriptor desc = new Descriptor(Directories.getBackupsDirectory(existing), KEYSPACE2, CF_STANDARD1, liveSSTable.descriptor.generation, liveSSTable.descriptor.formatType);
```

Comment:
Nit: can we break this line?
```suggestion
            Descriptor desc = new Descriptor(Directories.getBackupsDirectory(existing), 
                                             KEYSPACE2, 
                                             CF_STANDARD1, 
                                             liveSSTable.descriptor.generation, 
                                             liveSSTable.descriptor.formatType);
```